src/NovelRT/Input/InteractionObject.cpp

```diff
@@ -0,0 +1,19 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+#include <NovelRT.h>
+
+namespace NovelRT::Input {
+  InteractionObject::InteractionObject(const Transform& transform, int layer, const std::function<void(InteractionObject*)> notifyHasBeenDrawnObject)
+    : WorldObject(transform, layer) {
+    _notifyHasBeenDrawnObject = notifyHasBeenDrawnObject;
+  }
+
+  void InteractionObject::executeObjectBehaviour() {
+    _notifyHasBeenDrawnObject(this);
```

Comment:
what is the purpose of `notifyHasBeenDrawn`?

Reply:
This is how the service and the interaction rects talk to each other. The rect tells the service its been drawn, which triggers the service to manage if its been "clicked on", or not.